src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -3284,10 +3538,7 @@ public int parse(DateTimeParseContext context, CharSequence text, int position)
          * @throws DateTimeException if the value cannot be converted to a fraction
          */
         private BigDecimal convertToFraction(long value) {
-            ValueRange range = field.range();
-            range.checkValidValue(value, field);
-            BigDecimal minBD = BigDecimal.valueOf(range.getMinimum());
-            BigDecimal rangeBD = BigDecimal.valueOf(range.getMaximum()).subtract(minBD).add(BigDecimal.ONE);
```

Comment:
I wouldn't be surprised if there is a way to replace the use of `BigDecimal` with calculations using `long`.  Fundamentally, calculations like 15/60 -> 0.25 are not hard, but it depends on whether the exact results can be matched across a wide range of possible inputs.

Reply:
I think the main engineering challenge is writing tests that ensure that we don't lose precision on an arbitrary fractional field.